File: app/src/main/java/com/shindra/OurTrams/OurTramsAdapter.java
package com.shindra.OurTrams;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.shindra.R;

import java.util.ArrayList;

public class OurTramsAdapter extends RecyclerView.Adapter<OurTramsViewHolder>{

    // Attributes
    private ArrayList<OurTramsItem> mOurTramsList;
    private OnItemClickListener mListener;

    // Constructor
    public OurTramsAdapter(ArrayList<OurTramsItem> arrayList, OnItemClickListener listener) {
        mOurTramsList = arrayList;
        mListener = listener;
    }

    // Methods
    @NonNull
    @Override
    public OurTramsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        // Called to create each ViewHolder in RecyclerView
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.cv_our_trams, parent, false);
        OurTramsViewHolder tramVH = new OurTramsViewHolder(v);
        return tramVH;
    }
    @Override
    public void onBindViewHolder(@NonNull OurTramsViewHolder holder, int position) {
        // Called after onCreateViewHolder, to bind data for each ViewHolder in RecyclerView
        OurTramsItem currentItem = mOurTramsList.get(position);
        holder.onBind(currentItem, mListener);
    }
    @Override
    public int getItemCount() {
        return mOurTramsList.size();
    }
    public void SetOurTramsList(ArrayList<OurTramsItem> list)
    {
        mOurTramsList = list;
    }

    // Interface to handle event (observer pattern)
    public interface OnItemClickListener{
        void onButtonScheduleClick(String tramLineName);
    }
}

File: app/src/main/java/com/shindra/ourTrams/OurTramsAdapter.kt
package com.shindra.ourTrams

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.shindra.R

class OurTramsAdapter(private var mOurTramsList: ArrayList<OurTramsItem>, private var mListener: OnItemClickListener) : RecyclerView.Adapter<OurTramsViewHolder>() {

    // Methods
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OurTramsViewHolder {
        // Called to create each ViewHolder in RecyclerView
        val view = LayoutInflater.from(parent.context).inflate(R.layout.cv_our_trams, parent, false)
        return OurTramsViewHolder(view)
    }

    override fun onBindViewHolder(holder: OurTramsViewHolder, position: Int) {
        // Called after onCreateViewHolder, to bind data for each ViewHolder in RecyclerView
        val item = mOurTramsList[position]
        holder.onBind(item, mListener)
    }

    override fun getItemCount(): Int {
        return mOurTramsList.size
    }

    fun setOurTramsList(list: ArrayList<OurTramsItem>)
    {
        // Update ViewHolders DataSet
        mOurTramsList = list
    }

    // Interface
    interface OnItemClickListener {
        // Interface to handle event (observer pattern)
        fun onButtonScheduleClick(tramLineName: String)
    }
}